### ISSUE TITLE
Server in progress

### ISSUE THREAD

**Issue by JorenC** [zond/dipact#182]
After mustering or being the last one to commit, the game is sent to the queue. Then, for a couple of minutes, nothing happens - the game stays in its old state and no feedback is given on the game state.

Can we think of a way that we can get an "adjucation" state?
E.g. when loading a game, when pressing "commit", or when deadline passed (if game open), or even send Diplicity server push message to all players that the new turn will be adjucated (hidden in background, not player visible)?
During this state, the map and order screen would show some "waiting state" and check every minute or regular interval (or wait for adjucation complete) to see if the game has progressed.

Especially for live or F2F games, you want to have a live countdown (and then the adjucation time is important).

Now, when progressing a game, the only solution is a refresh, which feels like a bug, is questionable and harder to do on Android, and experienced quite often (1/7 players per turn on classic, if they are the last one to commit).

Maybe needs more design document, but let's chat about possibilities.

**Comment by zond** [comment]
That phase, like all others, reloads the game automatically when the notification arrives.

I don't know what this issue is actually about. Non-reloading games when phases resolve, or lack of countdown timer, or something else?

**Comment by JorenC** [comment]
Right now, moving to next turn reloads the game view? Then it's a bug, cause that never happened...

I press commit, even when I'm the last, and it takes a minute or so for the notification to arrive (adjucation queue)... But during that minute I'm in Limbo - in 1v1, I know the turn should move... That limbo is what I want to prevent, especially if we want to have the timer become seconds, so you can see it count down to zero and see the "waiting adjucation" state

**Comment by zond** [comment]
> in 1v1, I know the turn should move.

How do you know that?

If we have a timer it's of course easy-ish to reload when it reaches zero

**Comment by JorenC** [comment]
Because I get a notification about 30 seconds later, and I know AceBone did
his turn 10 hours ago ..

**Comment by zond** [comment]
So, for the typical usecase you _don't_ know that Acebone did his turn 10 hours ago, and those 30 seconds of waiting are _fine_ and a part of the game for most of the players in most of the games, when they wait for the turn to end... so why is it such a problem here?

**Comment by JorenC** [comment]
Maybe its not. But I had this issue quite a couple of times earlier, where
I was stuck and needed to reload - I even posted some issues in discord, I
think.

A reload is easy on web, but not possible on the APK (need to close and
reopen the game, which is not a natural or intuitive state).

I can gather more evidence before we make a decision, but I want to prevent
people from experiencing a "you need to restart the app" bug relatively
commonly... When they don't have the developer next to them and they don't
know that closing and reopening the game reloads it (rather than for
example switching map and order screen)...

**Comment by zond** [comment]
Yeah, it's true - especially in environments where you don't get notifications (like iOS)...

We could maybe use some kind of long polling of game state - I recently built it for chat channels when the notifications aren't supported - but too much long polling will load the server in weird ways I believe.

**Comment by JorenC** [comment]
Yes, you understand the issue.

I'm not saying it needs to be fixed right away, but if we can make a
solution that ALSO works with second counters (in one go), then I think we
have a new feature (F2F live game on a big screen, everyone can bring their
mobile device for breakout negotiations and orders. We could even add a
special "Spectator mode" where we zoom the screen correctly and also do
something in the way orders are displayed, just brainstorming

**Comment by zond** [comment]
The countdown thing is a completely separate solution though - it's just a timer, whereas the other one is a messaging problem (and they are always hard) - even if they both are about users discovering that the phase has ended.

**Comment by JorenC** [comment]
Sure. Me talking solutions is just a brain dump, it's the feature that I
think is needed in the long run.
